File: src/ProjectR.Scripting/content/scripts/spells/AlphaSlash.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Alpha Slash"
end

function GetDescription()
    return "A strong physical sword slash."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .4
end

function GetDelay()
  return .4
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {}
end

function SpellEffect()
  local damage = 6 * aAD - dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/AlphaSlash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class AlphaSlash : SpellScriptBase
    {
        public override string Name { get { return "Alpha Slash"; } }
        public override string Description { get { return "A strong physical sword slash."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] {  }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 40; } }
        public override double Delay { get { return .4; } }

        public override void SpellEffect(ICharacter caster, ICharacter target, double decayMod = 1)
        {
            var damage = 6 * AD(caster) - DEF(target);
            DealDamage(target, damage);
        }
    }
}